librz/core/cbin.c

```diff
@@ -1284,105 +1296,91 @@ RZ_API bool rz_core_bin_apply_imports(RzCore *core, RzBinFile *binfile, bool va)
 	return true;
 }
 
-static const char *get_prefix_for_sym(RzBinSymbol *sym) {
-	if (sym) {
-		// workaround for ELF
-		if (sym->type) {
-			if (!strcmp(sym->type, RZ_BIN_TYPE_NOTYPE_STR)) {
-				return sym->is_imported ? "loc.imp" : "loc";
-			}
-			if (!strcmp(sym->type, RZ_BIN_TYPE_OBJECT_STR)) {
-				return sym->is_imported ? "obj.imp" : "obj";
-			}
+static const char *get_prefix_for_sym(const RzBinSymbol *sym) {
+	if (!sym) {
+		return "sym";
+	}
+	// workaround for ELF
+	if (sym->type) {
+		if (!strcmp(sym->type, RZ_BIN_TYPE_NOTYPE_STR)) {
+			return sym->is_imported ? "loc.imp" : "loc";
+		}
+		if (!strcmp(sym->type, RZ_BIN_TYPE_OBJECT_STR)) {
+			return sym->is_imported ? "obj.imp" : "obj";
 		}
-		return sym->is_imported ? "sym.imp" : "sym";
 	}
-	return "sym";
+	return sym->is_imported ? "sym.imp" : "sym";
 }
 
-#define MAXFLAG_LEN_DEFAULT 128
+static char *construct_symbol_flagname(const RzBinSymbol *sym, const char *name) {
+	const char *prefix = get_prefix_for_sym(sym);
+	const char *lib = sym->libname ? sym->libname : "";
+	const char *separator = sym->libname ? "_" : "";
 
-static char *construct_symbol_flagname(const char *pfx, const char *libname, const char *symname, int len) {
-	char *r = rz_str_newf("%s.%s%s%s", pfx, libname ? libname : "", libname ? "_" : "", symname);
-	if (!r) {
+	if (!name) {
+		return NULL;
+	}
+
+	char *flagname = rz_str_newf("%s.%s%s%s", prefix, lib, separator, name);
+	if (!flagname) {
 		return NULL;
 	}
-	rz_name_filter(r, len, true); // maybe unnecessary..
-	char *R = __filterQuotedShell(r);
-	free(r);
-	return R;
+	rz_name_filter(flagname, -1, true);
+
+	// TODO: probably we should remove this due to newshell.
```

Comment:
@ret2libc take a look at this this thing in particular, please

Reply:
i doubt that now we suffer of code injections with newshell like we did before. but needs to be confirmed

Reply:
Yes we can likely remove this, all the names can be wrapped in quotes or double quotes or bad chars can be escaped in rzshell, so it should be fine. You could also try to remove it in separate PR to see how things change in a more manageable way.

Reply:
i think i will do this in another PR